packages/versioning/src/versioning.ts

```diff
@@ -94,12 +95,22 @@ export function $renamedFrom(context: DecoratorContext, t: Type, v: EnumMember,
   program.stateMap(renamedFromKey).set(t, record);
 }
 
-export function $madeOptional(context: DecoratorContext, t: Type, v: EnumMember) {
+export function $madeOptional(context: DecoratorContext, t: ModelProperty, v: EnumMember) {
   const { program } = context;
   const version = checkIsVersion(context.program, v, context.getArgumentTarget(0)!);
   if (!version) {
     return;
   }
+  if (!t.optional) {
```

Comment:
I can't find a repro right now but I feel like this is going to emit a diagnostic if you are projecting the older version when this is required.

Reply:
It is probably not happening right now because the decorator still runs on the projected type before the projection makes it required but I think if we run yet another projection on top of that then it will start to complain.

Reply:
A better way to validate that might be just to have that in the `onValidate` with the other versioning validation

Reply:
It seems so much more complicated to do this inside of onValidate. It doesn't raise a diagnostic when projecting the older required version and projects it correctly as required.

Reply:
Ok so this doesn't complain because I think the projection don't do what is expected.


If you have

```
const original = compile();
const projected1 = projectProgram(orginal);
const projected2 = projectProgram(projected1);
```

you'd expect the projection2 to run on top of the already proejcted program but it actually still runs on top of the original program

Reply:
so I guess if that's expected then this is not a problem and we can keep the validation like that.

Reply:
Relocated the logic into `onValidate`.